pyccel/codegen/printing/cwrappercode.py

```diff
@@ -190,35 +189,128 @@ def get_default_assign(self, arg, func_arg, value):
+        needed to create the function signature used for C/fortran binding
+
+        Parameters:
+        ----------
+        expr : FunctionDef
+            The function defintion
+
+        Return:
+        ------
+        String
+            Signature of the function
+        """
+        #if target_language is C no need for the binding
+        if self._target_language == 'c':
+            return self.function_signature(expr)
+
+        args = [a.var for a in expr.arguments]
+        if len(expr.results) == 1:
+            ret_type = self.get_declare_type(expr.results[0])
+        elif len(expr.results) > 1:
+            ret_type = self._print(datatype('int')) + ' '
+            args += [a.clone(name = a.name, is_pointer =True) for a in expr.results]
+        else:
+            ret_type = self._print(datatype('void')) + ' '
+        name = expr.name
+        if not args:
+            arg_code = 'void'
+        else:
+            arg_code = ', '.join('{}'.format(self.function_signature(i, False))
+                        if isinstance(i, FunctionAddress)
+                        else '{0}{1}'.format(self.get_static_declare_type(i), i.name)
+                        for i in args)
+
+        if isinstance(expr, FunctionAddress):
+            return '{}(*{})({})'.format(ret_type, name, arg_code)
+        else:
+            return '{0}{1}({2})'.format(ret_type, name, arg_code)
+
+    def get_static_args(self, argument):
+        """
+        Create bind_C arguments for arguments rank > 0 in fortran.
+        needed in static function call
+        func(a) ==> static_func(nd_dim(a) , nd_data(a))
+        where nd_data(a) = buffer holding data
+              nd_dim(a)  = size of array
+
+        Parameters
+        ----------
+        argument    : Variable
+            Variable holding information needed (rank)
+
+        Returns     : List of arguments
+            List that can contains Variables and FunctionCalls
+        -----------
+        """
+
+        if self._target_language == 'fortran' and argument.rank > 0:
+            arg_address = VariableAddress(argument)
+            static_args = [
+                FunctionCall(array_get_dim, [arg_address, i]) for i in range(argument.rank)
+            ]
+            static_args.append(FunctionCall(array_get_data, [arg_address]))
         else:
-            static_function = function
-            static_args = [a.var for a in function.arguments]
-        return static_function, static_args, additional_body
+            static_args = [argument]
+
+        return static_args
+
+    def get_static_declare_type(self, variable):
+        """
+        Get the declaration type of a variable, this function is used for
+        C/fortran binding using native C datatypes.
+
+        Parameters
+        ----------
+        variable : Variable
+            Variable holding information needed to choose the declaration type
+
+        Returns: String
+        --------
```

Comment:
Wrong formatting: `String` should be listed below the `---------` separator.

Reply:
Fixed : 23851c9